checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallInference.java

```diff
@@ -0,0 +1,837 @@
+      MethodInvocationNode invocation,
+      Node arg) {
+
+    for (int i = 1; i < paramsOfCurrentMethod.size() + 1; i++) {
+
+      VariableTree currentMethodParamTree = paramsOfCurrentMethod.get(i - 1);
+      VariableElement currentMethodParamElt =
+          TreeUtils.elementFromDeclaration(currentMethodParamTree);
+      if (isParamAndArgAliased(obligations, arg, currentMethodParamElt)) {
+
+        JavaExpression target = JavaExpression.fromVariableTree(currentMethodParamTree);
+        if (mustCallObligationSatisfied(invocation, currentMethodParamElt, target)) {
+          addOwningToParam(i);
+          break;
+        }
+      }
+    }
+  }
+
+  /**
+   * Returns the set of resource aliases associated with the given argument node, by looking up the
+   * corresponding obligation in the given set of obligations.
+   *
+   * @param obligations the set of obligations to search in
+   * @param arg the argument node whose resource aliases are to be returned
+   * @return the resource aliases associated with the given argument node, or an empty set if the
+   *     node has none
+   */
+  private Set<ResourceAlias> getResourceAliasOfArgument(Set<Obligation> obligations, Node arg) {
+    Node tempVar = mcca.getTempVarOrNode(arg);
+    if (!(tempVar instanceof LocalVariableNode)) {
+      return Collections.emptySet();
+    }
+
+    Obligation argumentObligation =
+        MustCallConsistencyAnalyzer.getObligationForVar(obligations, (LocalVariableNode) tempVar);
+    if (argumentObligation == null) {
+      return Collections.emptySet();
+    }
+    return argumentObligation.resourceAliases;
+  }
+
+  /**
+   * This method performs three checks related to method invocation node. It computes @Owning
+   * annotations to the enclosing formal parameter or fields:
+   *
+   * <ul>
+   *   <li>If a formal method is passed as an owning parameter, it adds the @Owning annotation to
+   *       that formal parameter (see {@link #analyzeOwnershipTransferAtMethodCall}).
+   *   <li>It calls {@link #analyzeOwnershipOfReceiverFromMethodInvocation} to verify if the
+   *       receiver of the method represented by {@code invocation} qualifies as a candidate owning
+   *       field, and if the method invocation satisfies the field's must-call obligation. If these
+   *       conditions are met, the field is added to the {@link #releasedFields} set.
+   *   <li>It calls {@link #analyzeCalledMethodsOfInvocationArgs} to inspect the method represented
+   *       by the given MethodInvocationNode for any indirect calls within it. The method analyzes
+   *       the called-methods set of each argument after the call and computes the @Owning
+   *       annotation to the field or parameter passed as an argument to this call.
+   * </ul>
+   *
+   * @param obligations the set of obligations to search in
+   * @param invocation the MethodInvocationNode
+   */
+  private void checkMethodInvocation(Set<Obligation> obligations, MethodInvocationNode invocation) {
```

Comment:
"Check" is not very descriptive.  Should this be named "computeOwningFromMethodInvocation" or the like?